### PR TITLE
Add Event Grid Blob Storage trigger templates

### PR DESCRIPTION
**Pull request by castrodd**
This PR is adding templates for the Event Grid-based Blob Storage trigger (as described in [this tutorial](https://learn.microsoft.com/en-us/azure/azure-functions/functions-event-grid-blob-trigger?tabs=isolated-process%2Cnodejs-v4&pivots=programming-language-csharp)).

What's included:
- Templates for C# (isolated), Python, JavaScript/TypeScript (old model), and PowerShell

What's _not_ included:
- Templates for in-process C#, F#, Java, JavaScript/TypeScript (new model)
    - In-process C# not supported on Flex, so deprioritizing those templates
    - F# is not listed in the above tutorial, so not including in this first wave of sample templates
    - Java templates are not included in this repository
      - Java template also does not seem to be working. Still need to verify it actually works. 
    - New model JS/TS templates are not included in this repository

### CODE REVIEW

**Review by mattchenderson — CHANGES_REQUESTED** [review]
Missing template config, which is required for .NET

File: Build/PackageFiles/Dotnet_precompiled/ItemTemplates-Isolated_v3.x.nuspec [hunk 1]
```diff
@@ -27,5 +27,6 @@
     <file src="..\..\..\Functions.Templates\Templates\CosmosDBTrigger-CSharp-Isolated\**" target="content\CosmosDBTrigger-CSharp" />
     <file src="..\..\..\Functions.Templates\Templates\SignalRConnectionInfoHttpTrigger-CSharp-Isolated\**" target="content\SignalRConnectionInfoHttpTrigger-CSharp" />
     <file src="..\..\..\Functions.Templates\Templates\RabbitMQTrigger-CSharp-Isolated\**" target="content\RabbitMQTrigger-CSharp" />
+    <file src="..\..\..\Functions.Templates\Templates\EventGridBlobTrigger-CSharp-Isolated\**" target="content\EventGridBlobTrigger-CSharp" />
```
Comment (soninaren) [inline Build/PackageFiles/Dotnet_precompiled/ItemTemplates-Isolated_v3.x.nuspec:30]: This package is not used anymore, Only need to update 
Build/PackageFiles/Dotnet_precompiled/ItemTemplates-Isolated_v4.x.nuspec

File: Build/PackageFiles/Dotnet_precompiled/ItemTemplates_v3.x.nuspec [hunk 2]
```diff
@@ -45,5 +45,6 @@
     <file src="..\..\..\Functions.Templates\Templates\KafkaOutput-CSharp\**" target="content\KafkaOutput-CSharp" exclude="**\*.csx;**\*.md;**\*.dat;**\function.json;**\metadata.json" />
     <file src="..\..\..\Functions.Templates\Templates\SqlInputBinding-CSharp\**" target="content\SqlInputBinding-CSharp" exclude="**\*.csx;**\*.md;**\*.dat;**\function.json;**\metadata.json" />
     <file src="..\..\..\Functions.Templates\Templates\SqlOutputBinding-CSharp\**" target="content\SqlOutputBinding-CSharp" exclude="**\*.csx;**\*.md;**\*.dat;**\function.json;**\metadata.json" />
+    <file src="..\..\..\Functions.Templates\Templates\EventGridBlobTrigger-CSharp-Isolated\**" target="content\EventGridBlobTrigger-CSharp" />
```
Comment (soninaren) [inline Build/PackageFiles/Dotnet_precompiled/ItemTemplates_v3.x.nuspec:48]: Same here, only need to update ItemTemplates_v4

File: Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
+{
+    "author": "Microsoft",
+    "classifications": [
+        "Azure Function",
+        "Trigger",
+        "Blob",
+        "EventGrid"
+    ],
+    "name": "EventGridBlobTrigger",
+    "identity": "Azure.Function.CSharp.Isolated.EventGridBlobTrigger.3.x",
```
Comment (soninaren) [inline Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json:10]:
```suggestion
    "identity": "Azure.Function.CSharp.Isolated.EventGridBlobTrigger.6.x",
```

File: Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
+{
```
Comment (soninaren) [inline Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json:1]: Can you rename the directory to `EventGridBlobTrigger-CSharp-Isolated-6.x`

File: Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    },
+    "sourceName": "EventGridBlobTriggerCSharp",
+    "symbols": {
+        "namespace": {
+            "description": "namespace for the generated code",
+            "replaces": "Company.Function",
+            "type": "parameter"
+        },
+        "Connection": {
+            "description": "The name of the app setting containing your Storage connection.",
```
Comment (soninaren) [inline Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json:25]:
```suggestion
            "description": "The name of the app setting containing your storage connection.",
```

File: Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
+{
+    "author": "Microsoft",
+    "classifications": [
+        "Azure Function",
+        "Trigger",
+        "Blob",
+        "EventGrid"
```
Comment (soninaren) [inline Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json:7]: I would suggest removing event grid if it is not directly using event grid or taking event grid config as part of the template

File: Functions.Templates/Templates/EventGridBlobTrigger-JavaScript/function.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (soninaren) [inline Functions.Templates/Templates/EventGridBlobTrigger-JavaScript/function.json:1]: `Functions.Templates/Templates/EventGridBlobTrigger-JavaScript` is missing metadata.json file

File: Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    "postActions": [
+        {
+            "Description": "Adding Reference to Microsoft.Azure.Functions.Worker.Extensions.Storage.Blobs Nuget package",
+            "ActionId": "B17581D1-C5C9-4489-8F0A-004BE667B814",
+            "ContinueOnError": "true",
+            "ManualInstructions": [],
+            "args": {
+                "referenceType": "package",
+                "reference": "Microsoft.Azure.Functions.Worker.Extensions.Storage.Blobs",
+                "version": "6.3.0",
```
Comment (soninaren) [inline Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json:56]: 6.4.0 is out. Can use that one now.
```suggestion
                "version": "6.4.0",
```

File: Functions.Templates/Templates/EventGridBlobTrigger-JavaScript/function.json (hunk 7 above)
Comment (soninaren) [inline Functions.Templates/Templates/EventGridBlobTrigger-JavaScript/function.json:1]: add the template to `/Build/PackageFiles/ExtensionBundle/ExtensionBundleTemplates-4.x.nuspec`

File: Functions.Templates/Templates/EventGridBlobTrigger-PowerShell/function.json [hunk 9]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (soninaren) [inline Functions.Templates/Templates/EventGridBlobTrigger-PowerShell/function.json:1]: add the template to `/Build/PackageFiles/ExtensionBundle/ExtensionBundleTemplates-4.x.nuspec`

File: Functions.Templates/Templates/EventGridBlobTrigger-Python/__init__.py [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import logging
```
Comment (soninaren) [inline Functions.Templates/Templates/EventGridBlobTrigger-Python/__init__.py:1]: add the template to `/Build/PackageFiles/ExtensionBundle/ExtensionBundleTemplates-4.x.nuspec`

**Review by soninaren — CHANGES_REQUESTED** [review]
Have requested some minor changes. Looks good otherwise.

File: Functions.Templates/Templates/EventGridBlobTrigger-JavaScript/function.json (hunk 7 above)
Comment (castrodd) [reply Functions.Templates/Templates/EventGridBlobTrigger-JavaScript/function.json:1]: Added

File: Functions.Templates/Templates/EventGridBlobTrigger-PowerShell/function.json (hunk 9 above)
Comment (castrodd) [reply Functions.Templates/Templates/EventGridBlobTrigger-PowerShell/function.json:1]: Added

File: Functions.Templates/Templates/EventGridBlobTrigger-Python/__init__.py (hunk 10 above)
Comment (castrodd) [reply Functions.Templates/Templates/EventGridBlobTrigger-Python/__init__.py:1]: Added

File: Build/PackageFiles/Dotnet_precompiled/ItemTemplates-Isolated_v3.x.nuspec (hunk 1 above)
Comment (castrodd) [reply Build/PackageFiles/Dotnet_precompiled/ItemTemplates-Isolated_v3.x.nuspec]: Removed

File: Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json (hunk 3 above)
Comment (castrodd) [reply Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json]: Updated

File: Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json (hunk 5 above)
Comment (castrodd) [reply Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json]: Updated

File: Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json (hunk 8 above)
Comment (castrodd) [reply Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json]: Updated

File: Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json (hunk 6 above)
Comment (castrodd) [reply Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json]: Removed

File: Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json (hunk 4 above)
Comment (castrodd) [reply Functions.Templates/Templates/EventGridBlobTrigger-CSharp-Isolated/.template.config/template.json]: Renamed

File: Build/PackageFiles/Dotnet_precompiled/ItemTemplates_v3.x.nuspec (hunk 2 above)
Comment (castrodd) [reply Build/PackageFiles/Dotnet_precompiled/ItemTemplates_v3.x.nuspec]: Removed

**Review by soninaren — APPROVED** [review]

**Pull request merged**